### DISCUSSION TITLE
How can I use setStartSec() ?

### DISCUSSION THREAD

**Discussion by vintagecardriver** [pschatzmann/arduino-audio-tools#1619] [Q&A] [answered]
Hello,
when I'm trying to start a song at a predefined moment for example 50 seconds after the beginning of the song with "setStartSec(50);", I'm getting the fault code: 'setStartSec()' was not declared in this scope.
How can I use "setStartSec()" in the right way?

**Comment by pschatzmann** [comment, marked as answer]
Just send the PCM output to a [TimedStream](https://pschatzmann.github.io/arduino-audio-tools/classaudio__tools_1_1_timed_stream.html)

**Comment by vintagecardriver** [comment]
I'm sorry, but I'm not an experienced programmer. 
Would you be so kind, please and just give a short example, how to use TimedStream in a sketch, when song with index 33 (33. File on the SD Card) should start 50 seconds after beginning of the song?
I've tried 

'TimedStream(AudioOutput &o, 50, -1); 
player.setIndex(33);' 

but hat no success. The fault message is: expected primary-expression before '(' token

I've also tried:

TimedStream timed (source.index(), 50, -1)

and the error message is no matching function for call to 'audio_tools::TimedStream::TimedStream(int, int, int)'


Perhaps you could add the TimedStream call please in your sd-i2C-example for understanding how the TimedStream will work. Where and how could I add the TimedStream call in that code?

You would make me very happy with a little help!

/**
 * @file player-sd-i2s.ino
 * @brief example using the SD library
 * 
 * @author Phil Schatzmann
 * @copyright GPLv3
 */

#include "AudioTools.h"
#include "AudioLibs/AudioSourceSD.h"
#include "AudioCodecs/CodecMP3Helix.h"

const char *startFilePath="/";
const char* ext="mp3";
AudioSourceSD source(startFilePath, ext);
I2SStream i2s;
MP3DecoderHelix decoder;
AudioPlayer player(source, i2s, decoder);

void setup() {
  Serial.begin(115200);
  AudioLogger::instance().begin(Serial, AudioLogger::Info);

  // setup output
  auto cfg = i2s.defaultConfig(TX_MODE);
  i2s.begin(cfg);

  // setup player
  player.begin();
}

void loop() {
  player.copy();
}

**Reply by vintagecardriver** [reply]
Hello Phil,

would you be so kind, please and give a little help, how to use the TimedStream call?

Thank you very much and best regards
